### PR TITLE
Added VichUploaderBundle recipe

### PR DESCRIPTION
**Pull request by javiereguiluz**
About the bundle:

* The most popular one for file uploading
* 1.2 million downloads
* https://github.com/dustin10/VichUploaderBundle

License MIT

### CODE REVIEW

**Comment by javiereguiluz** [comment]
In the original bundle docs, they use `%kernel.root_dir%/../` instead of `%kernel.project_dir%/` but I've changed it to align with modern Symfony practices. Should Symfony Bot check if `kernel.root_dir` is used and suggest a change?

**Comment by Pierstoval** [comment]
Actually, it would be better if the bot could detect `%kernel.root_dir%/../` instead IMO

**Comment by yceruto** [comment]
@fabpot I've tested this recipe and I don't see the `Setting configuration and copying files` step:
```sh
Package operations: 11 installs, 0 updates, 0 removals
  - Installing jms/metadata (1.6.0): Loading from cache
  - Installing symfony/asset (dev-master 67ee69a): Loading from cache
  - Installing symfony/inflector (dev-master aed5a08): Loading from cache
  - Installing paragonie/random_compat (v2.0.10): Loading from cache
  - Installing symfony/polyfill-php70 (dev-master 13ce343): Loading from cache
  - Installing symfony/property-access (dev-master 1bb34c3): Loading from cache
  - Installing symfony/polyfill-util (dev-master 746bce0): Loading from cache
  - Installing symfony/polyfill-php56 (dev-master 1dd42b9): Loading from cache
  - Installing symfony/security (dev-master 39263c7): Downloading (100%)         
  - Installing symfony/templating (dev-master 503b8be): Loading from cache
  - Installing vich/uploader-bundle (dev-master 83c3e27): Downloading (100%)         
    Detected auto-configuration settings for "vich/uploader-bundle"
    Enabling the package as a Symfony bundle
```
No file copied.

**Comment by javiereguiluz** [comment]
@yceruto in case you did the same mistake as I did, have you enabled the installation of contrib recipes?  `composer config extra.symfony.allow-contrib true`

**Comment by yceruto** [comment]
@javiereguiluz yeap :) all of them https://symfony.sh/r/contrib/22

**Comment by yceruto** [comment]
Hence, if `cli` has been already installed before, while `composer req "vich/uploader-bundle:^1.5"`:
```sh
...
Executing script assets:install --symlink --relative %WEB_DIR% [KO]
 [KO]
Script assets:install --symlink --relative %WEB_DIR% returned with error code 1
!!  
!!  
!!                                                                                   
!!    [Symfony\Component\Config\Definition\Exception\InvalidConfigurationException]  
!!    The child node "db_driver" at path "vich_uploader" must be configured.         
!!  
!!                                                                                   
!!  
!!  
!!  

Installation failed, reverting ./composer.json to its original content.
```

**Comment by Pierstoval** [comment]
@yceruto Making `vich/uploader-bundle` require `symfony/console` and might solve the issue

**Comment by yceruto** [comment]
@Pierstoval I don't think so, the issue is that https://github.com/symfony/recipes-contrib/pull/22#issuecomment-301796198

**Comment by fabpot** [comment]
@yceruto That's because it installed the master version (which is 1.6) and this recipe is for 1.5 only.

**Comment by yceruto** [comment]
@fabpot same issue for 1.5:
```sh
$ composer req "vich/uploader-bundle:1.5"
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 15 installs, 0 updates, 0 removals
  - Installing ocramius/package-versions (1.1.2): Loading from cache
  - Installing jms/metadata (1.6.0): Loading from cache
  - Installing zendframework/zend-eventmanager (dev-develop 42d2abc): Loading from cache
  - Installing zendframework/zend-code (dev-develop e8e6491): Loading from cache
  - Installing ocramius/proxy-manager (2.1.x-dev e18ac87): Loading from cache
  - Installing symfony/asset (dev-master 67ee69a): Loading from cache
  - Installing symfony/inflector (dev-master aed5a08): Loading from cache
  - Installing paragonie/random_compat (v2.0.10): Loading from cache
  - Installing symfony/polyfill-php70 (dev-master 13ce343): Loading from cache
  - Installing symfony/property-access (dev-master 1bb34c3): Loading from cache
  - Installing symfony/polyfill-util (dev-master 746bce0): Loading from cache
  - Installing symfony/polyfill-php56 (dev-master 1dd42b9): Loading from cache
  - Installing symfony/security (dev-master 39263c7): Loading from cache
  - Installing symfony/templating (dev-master 503b8be): Loading from cache
  - Installing vich/uploader-bundle (1.5.0): Loading from cache
    Detected auto-configuration settings for "vich/uploader-bundle"
    Enabling the package as a Symfony bundle
Writing lock file
Generating autoload files
ocramius/package-versions:  Generating version class...
ocramius/package-versions: ...done generating version class
Executing script make cache-warmup [OK]
Executing script assets:install --symlink --relative %WEB_DIR% [KO]
 [KO]
Script assets:install --symlink --relative %WEB_DIR% returned with error code 1
!!  
!!  
!!                                                                                   
!!    [Symfony\Component\Config\Definition\Exception\InvalidConfigurationException]  
!!    The child node "db_driver" at path "vich_uploader" must be configured.         
!!  
!!                                                                                   
!!  
!!  
!!  

Installation failed, reverting ./composer.json to its original content.
```

**Comment by yceruto** [comment]
Somebody else have the same issue or am I the only one?

**Comment by Koc** [comment]
@yceruto let's wait until 1.6 version of bundle will released

**Comment by fabpot** [comment]
@yceruto Try using the latest version 1.5.3, and it works:

```
composer req "vich/uploader-bundle:1.5.3"
```

**Comment by yceruto** [comment]
I had to clean the composer cache to make it work, I don't know why :(

On the other hand, this bundle require `doctrine/annotations`:
```sh
Executing script assets:install --symlink --relative %WEB_DIR% [KO]
 [KO]
Script assets:install --symlink --relative %WEB_DIR% returned with error code 1
!!  
!!  
!!                                                                                 
!!  
!!  
!!    [Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException]   
!!  
!!  
!!    The service "vich_uploader.property_mapping_factory" has a dependency on a   
!!  
!!  
!!    non-existent service "annotation_reader".                                    
!!  
!!  
!!                                                                                 
!!  
!!  
!!  
!!  
!!  

Installation failed, reverting ./composer.json to its original content.
```

**Comment by stof** [comment]
@yceruto FrameworkBundle does not install the annotation layer by default anymore in 3.3. If the bundle needs it, it should add a dependency on `doctrine/annotations`

File: vich/uploader-bundle/1.5/etc/packages/vich_uploader.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+vich_uploader:
+    db_driver: orm
+
+    # mappings:
```
Comment (fabpot) [inline vich/uploader-bundle/1.5/etc/packages/vich_uploader.yaml]: I agree, people will just remove the `#` and the indentation will be off.

**Pull request merged**